src/playground-service-worker-proxy.ts

```diff
@@ -46,6 +51,19 @@ import {
     {scope}
   );
 
+  window.addEventListener(
+    'message',
+    (event: MessageEvent<PlaygroundMessage>) => {
+      if (event.data.type === UPDATE_SERVICE_WORKER) {
+        // When the project handshakes with the service worker, it may notice a
+        // version mismatch, in which case it will send this message to request
+        // the service worker update. Note that service workers eventually
+        // update automatically, but not necessarily right away.
+        registration.update();
```

Comment:
Is there any result to wait for here? Should the main window elements show some affordance that the SW is updating, or an error if it fails to?

Reply:
> Is there any result to wait for here? Should the main window elements show some affordance that the SW is updating, or an error if it fails to?

We should probably show an error if we don't eventually get a new SW after calling this. However, we don't currently have a place to show *any* failure states, of which there are are a bunch. Seems like something to address in the future -- maybe we'll want an error notification element or something. Added a TODO for now.